src/newcharacter.cpp

```diff
@@ -115,25 +115,83 @@ static bool isWide = false;
 // The point after which stats cost double
 static constexpr int HIGH_STAT = 12;
 
-// The ID of the rightmost tab
-static constexpr int NEWCHAR_TAB_MAX = 7;
-
 static int skill_increment_cost( const Character &u, const skill_id &skill );
 
-enum struct tab_direction {
-    NONE,
-    FORWARD,
-    BACKWARD,
-    QUIT
-};
-
 enum class pool_type {
     FREEFORM = 0,
     ONE_POOL,
     MULTI_POOL,
     TRANSFER,
 };
 
+
+class tab_manager
+{
+        std::vector<std::string> &tab_names;
+        std::map<size_t, inclusive_rectangle<point>> tab_map;
+        point window_pos;
+    public:
+        bool complete = false;
+        bool quit = false;
+        tab_list position;
+
+        explicit tab_manager( std::vector<std::string> &tab_names ) : tab_names( tab_names ),
+            position( tab_list( tab_names ) ) {
```

Comment:
This looks like you're needlessly construction a temporary object (`tab_list( tab_names )`) that is used to initialize the member `position`. Is this necessary?
```suggestion
            position( tab_names ) {
```

Reply:
A quick test suggests that no, it isn't necessary.  Thanks for the note.